mobile/src/home-view.js

```diff
@@ -353,11 +354,11 @@ class HomeView extends Component {
             var newObject = data.val()
             newObject.score = score
             newObject.myVote = myVote
-            if (data.val().creator.id === client.currentUser.id && oldState !== data.val().approve){
+            if (data.val().creator.id === client.currentUser.id && oldState !== data.val().approve && !data.val().block && !data.val().answered){
```

Comment:
You have already grabbed `newObject = data.val()`.  No need to call `data.val()` 3 more times in this line.